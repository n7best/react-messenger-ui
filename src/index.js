import { render } from './messemger-renderer/render';
import CONSTANTS from './components/Messaging/constants';

export const version = '0.0.16';
export const Text = 'TEXT';
export const Message = 'MESSAGE';
export const SenderAction = 'SENDERACTION';
export const Attachment = 'ATTACHMENT';
export const QuickReply = 'QUICKREPLY';
export const Button = 'BUTTON';
export const URLButton = 'URLBUTTON';
export const PostbackButton = 'POSTBACKBUTTON';
export const ShareButton = 'SHAREBUTTON';
export const CallButton = 'CALLBUTTON';
export const LoginButton = 'LOGINBUTTON';
export const LogoutButton = 'LOGOUTBUTTON';
export const GameplayButton = 'GAMEPLAYBUTTON';
export const Template = 'TEMPLATE';
export const Element = 'ELEMENT';
export const GenericElement = 'GENERICELEMENT';
export const ButtonTemplate = 'BUTTONTEMPLATE';
export const GenericTemplate = 'GENERICTEMPLATE';
export const ListTemplate = 'LISTTEMPLATE';
export const ListElement = 'LISTELEMENT';
export const MediaTemplate = 'MEDIATEMPLATE';
export const MediaElement = 'MEDIAELEMENT';
export const Address = 'ADDRESS';
export const Adjustment = 'ADJUSTMENT';
export const Summary = 'SUMMARY';
export const ReceiptTemplate = 'RECEIPTTEMPLATE';
export const ReceiptElement = 'RECEIPTELEMENT';
export const OpenGraphTemplate = 'OPENGRAPHTEMPLATE';
export const OpenGraphElement = 'OPENGRAPHELEMENT';
export const PersistentMenu = 'PERSISTENTMENU';
export const Menu = 'MENU';
export const NestedMenu = 'NESTEDMENU';
export {
  render,
  CONSTANTS
};

const MessengerUI = {
  render,
  Text,
  Message,
  SenderAction,
  Attachment,
  QuickReply,
  Button,
  URLButton,
  PostbackButton,
  ShareButton,
  CallButton,
  LoginButton,
  LogoutButton,
  GameplayButton,
  Template,
  Element,
  GenericElement,
  ButtonTemplate,
  GenericTemplate,
  ListTemplate,
  ListElement,
  MediaTemplate,
  MediaElement,
  Address,
  Adjustment,
  ReceiptElement,
  ReceiptTemplate,
  Summary,
  OpenGraphTemplate,
  OpenGraphElement,
  PersistentMenu,
  Menu,
  NestedMenu,
  CONSTANTS,
  version
};

export default MessengerUI;
