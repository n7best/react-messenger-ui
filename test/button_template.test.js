import React from 'react';
import { render, Message, ButtonTemplate, URLButton, PostbackButton, ShareButton, CallButton } from '../src';

test('should render button tempalte response', () => {
  const App = () => (
    <Message recipient={{ id: '<PSID>' }}>
      <ButtonTemplate>
        hello
        <URLButton url="magic link">
          world
        </URLButton>
        <PostbackButton payload="magic payload">
          world
        </PostbackButton>
        <CallButton payload="+1234567890">
          call
        </CallButton>
      </ButtonTemplate>
    </Message>
  )

  expect(render(<App />)).toMatchSnapshot()
});

test('should throw since share is not support by button template', () => {
  const App = () => (
    <Message recipient={{ id: '<PSID>' }}>
      <ButtonTemplate>
        hello
        <URLButton url="magic link">
          world
        </URLButton>
        <PostbackButton payload="magic payload">
          world
        </PostbackButton>
        <ShareButton payload="magic share">
          world
        </ShareButton>
      </ButtonTemplate>
    </Message>
  )

  expect(()=>render(<App />)).toThrow()
});


