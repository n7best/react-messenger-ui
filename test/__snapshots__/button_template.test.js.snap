// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render button tempalte response 1`] = `
Object {
  "message": Object {
    "attachment": Object {
      "payload": Object {
        "buttons": Array [
          URLButton {
            "title": "world",
            "type": "web_url",
            "url": "magic link",
          },
          PostbackButton {
            "payload": "magic payload",
            "title": "world",
            "type": "postback",
          },
          CallButton {
            "payload": "+1234567890",
            "title": "call",
            "type": "phone_number",
          },
        ],
        "template_type": "button",
        "text": "hello",
      },
      "type": "template",
    },
  },
  "messaging_type": "RESPONSE",
  "recipient": Recipient {
    "id": "<PSID>",
  },
}
`;

exports[`should render button tempalte response with trim 1`] = `
Object {
  "message": Object {
    "attachment": Object {
      "payload": Object {
        "buttons": Array [
          URLButton {
            "title": "world",
            "type": "web_url",
            "url": "magic link",
          },
          PostbackButton {
            "payload": "magic payload",
            "title": "world",
            "type": "postback",
          },
          CallButton {
            "payload": "+1234567890",
            "title": "call",
            "type": "phone_number",
          },
        ],
        "template_type": "button",
        "text": "hello",
      },
      "type": "template",
    },
  },
  "messaging_type": "RESPONSE",
  "recipient": Recipient {
    "id": "<PSID>",
  },
}
`;
